Ignore --enable-sha2 if OpenSSL doesn't support SHA256.
This patch is a quick fix until pkgsrc's Makefile is improved.

--- configure.orig	2009-08-14 12:34:17.000000000 +0000
+++ configure
@@ -14304,19 +14304,18 @@
 
   LIBS="-lcrypto $LIBS"
 
+  cat >>confdefs.h <<_ACEOF
+#define USE_SHA2 1
+_ACEOF
+
 else
 
-            { { $as_echo "$as_me:$LINENO: error: SHA2 enabled, but no SHA2 functions found in OpenSSL" >&5
-$as_echo "$as_me: error: SHA2 enabled, but no SHA2 functions found in OpenSSL" >&2;}
-   { (exit 1); exit 1; }; }
+            { $as_echo "$as_me:$LINENO: WARNING: SHA2 enabled, but no SHA2 functions found in OpenSSL" >&5
+$as_echo "$as_me: WARNING: SHA2 enabled, but no SHA2 functions found in OpenSSL" >&2;}
 
 fi
 
 
-cat >>confdefs.h <<_ACEOF
-#define USE_SHA2 1
-_ACEOF
-
         ;;
     no|*)
         ;;
